sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -110,16 +113,17 @@ public void pointTo(Object baseObject, long baseOffset, int sizeInBytes) {
     this.baseObject = baseObject;
     this.baseOffset = baseOffset;
     this.sizeInBytes = sizeInBytes;
+    this.headerInBytes = calculateHeaderPortionInBytes(numElements);
```

Comment:
```numElements``` is extracted from ```baseObject``` that is passed by ```pointTo```. If the constructor can have an argument for ```numElements```, we could assign the value at the constructor.